test/infrastructure/docker/api/v1alpha3/dockermachinetemplate_webhook.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"errors"
+	"reflect"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+func (m *DockerMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(m).
+		Complete()
+}
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-dockermachinetemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=dockermachinetemplates,versions=v1alpha3,name=validation.dockermachinetemplate.infrastructure.cluster.x-k8s.io,sideEffects=None
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha3-dockermachinetemplate,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=dockermachinetemplates,versions=v1alpha3,name=default.dockermachinetemplate.infrastructure.cluster.x-k8s.io,sideEffects=None
+
+var _ webhook.Defaulter = &DockerMachineTemplate{}
+var _ webhook.Validator = &DockerMachineTemplate{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (m *DockerMachineTemplate) Default() {
+}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (m *DockerMachineTemplate) ValidateCreate() error {
+	var allErrs field.ErrorList
+	if len(m.Spec.Template.Spec.ExtraMounts) == 0 {
```

Comment:
Uhhmm...
AFAIK extra mount are required only in few use-cases, like e.g. if we want to use CAPD inside a container created by CAPD (self-hosting), but it is perfectly fine to not have extra mounts for all the other use cases...

Reply:
Thanks for clarifying this. Removed this check.